### PR TITLE
Adding the rename function in the ListSelectionActivity

### PR DESCRIPTION
**Pull request by shoutizix**
Hey guys !

This addresses the issue #66 I had to refactor a bit the code and implement an adapter like we used for the ItemViewActivity class. I have also changed the ScrollView that contained TextView by a ListView containing TextView since it was better to get which view was cliked on and modify/open the corresponding TodoList. 
I think that #59 should be pretty easy using the adapter I've added in this PR. 

I still have to add tests and use the Database but otherwise the rename functionality should work great !

### CODE REVIEW

**Comment by shoutizix** [comment]
Now all should be great !
I'm waiting for your reviews :)

File: app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java:81]: Shouldn't this ID be the list's ID? Something like `todoLists.get(i).getId()`.

File: app/src/main/res/layout/layout_todo_list_item.xml
Comment (YagoGG) [inline app/src/main/res/layout/layout_todo_list_item.xml:14]: Not super sure about this, but I believe that if `"TextView"` is just a placeholder, we can just remove the whole `android:text` attribute. That way we avoid useless string (that are also considered hardcoded by the linter).

**Review by YagoGG — CHANGES_REQUESTED** [review]
Hey @shoutizix! Thanks a lot for the PR.

I left a couple inline comments about the code.

As for the feature, it looks super neat! I have been able to try it out in my emulator and it worked like a charm. I found a small issue, though:

Whenever you try to rename a list, an alert pops up asking you for the new name. However, the input field allows multi line inputs, which leads to weird behaviour:
    image
    image
    image

**Comment by shoutizix** [comment]
Hey !

You're absolutely right on the points that you've mentioned, I've made the corresponding changes.

File: app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java
Comment (theo-damiani) [inline app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java:60]: Maybe it is better to use a recycler view and a holder. (better performance and more functionality)
And we can maybe put it in another file so it can be re-used if necessary and it's better for understanding the global structure of the app.

**Review by theo-damiani — APPROVED** [review]
Great work!

I just mentioned two little things, but for the first one (regarding the UUID instead of the title), I think we will address it when we will implement the view model for this activity, for now, without it, you can't really do differently

File: app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java:80]: Why are we using this instead of the ID? :thinking:

**Review by YagoGG — COMMENTED** [review]
Hey @shoutizix! Thanks a lot for the changes.

I think this is about to be ready to merge. I just have a question regarding the significant bits trick, and I also share with @theo-damiani the preference for `UUID`s instead of titles (which, unlike `UUID`s, might not be unique).

Cheers!

**Comment by YagoGG** [comment]
Please do not forget to update the branch with the latest `master` to make sure that there are no merge conflicts. Thanks a lot! :)

File: app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java
Comment (shoutizix) [reply app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java:80]: Because the ID is an int and I've made some research and it seemed a used way to do but we could certainly change that in the future

**Comment by shoutizix** [comment]
Yes sure sorry I resolve the conflict now it should be good to merge !

**Review by lgraziano — APPROVED** [review]
I read the code and tried it, it works perfectly ! Very nice job 👌

File: app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java
Comment (YagoGG) [reply app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java]: So just to be clear, is this to cast the ID from `int` to `long`? Because if that's the case, you can just do:

```java
(long) todoLists.get(i).getId()
```

Comment (shoutizix) [reply app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java]: well yes it would make sense and it's more understandable I'll do it rn

Comment (YagoGG) [reply app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java]: Makes sense, thanks for the clarification!

**Comment by theo-damiani** [comment]
Again maybe the alert dialog will cause some errors. But works fine for me.

File: app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java:55]: Looks like we're not using this anymore, maybe we can remove it?

**Review by YagoGG — APPROVED** [review]
Hey, @theo-damiani!

Thanks a lot for the changes; I just did a second batch of functional tests and the new feature now works like a charm.

I left a minor comment, and the linter is also complaining about a couple hardcoded strings & missing attributes. Not super important, but it'd be great if we could take care of those before merging. LGTM otherwise! :+1: 

Cheers!

**Pull request merged**
